### ISSUE TITLE
Cannot use Saltcheck module

### ISSUE THREAD

**Issue by kiemlicz** [saltstack/salt#56324]
### Description of Issue
`saltcheck` module throws
```
AttributeError: 'set' object has no attribute 'extend'             
Traceback (most recent call last):                                                                 
  File "/usr/bin/salt-call", line 11, in <module>                   
    load_entry_point('salt==3000', 'console_scripts', 'salt-call')()                           
  File "/usr/lib/python3/dist-packages/salt/scripts.py", line 445, in salt_call
    client.run()                                                                                       
  File "/usr/lib/python3/dist-packages/salt/cli/call.py", line 57, in run
    caller.run()                                      
  File "/usr/lib/python3/dist-packages/salt/cli/caller.py", line 119, in run
    ret = self.call()                            
  File "/usr/lib/python3/dist-packages/salt/cli/caller.py", line 218, in call
    ret['return'] = self.minion.executors[fname](self.opts, data, func, args, kwargs)
  File "/usr/lib/python3/dist-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)                                         
  File "/usr/lib/python3/dist-packages/salt/modules/saltcheck.py", line 430, in run_state_tests
    stl.add_test_files_for_sls(state_name, check_all)                       
  File "/usr/lib/python3/dist-packages/salt/modules/saltcheck.py", line 1112, in add_test_files_for_sls
    cached_copied_files.extend(this_cache_ret)                               
AttributeError: 'set' object has no attribute 'extend'  
```

I believe the bug is in `salt/modules/saltcheck.py:1136`:
```
        cached_copied_files = []
...
        for low_data in ret:
...
                   cached_copied_files = set(cached_copied_files) # type changes here, loop may continue
```
[Obligatory meme, the friend of mine shown me some time ago](https://pbs.twimg.com/media/B20YoFiIIAAHCYf?format=jpg&name=small)

### Setup
For following [State Tree](https://github.com/kiemlicz/ambassador/tree/master/salt) I was trying to write saltcheck test by placing `saltcheck-test` dir inside `salt/base/os/` directory

### Steps to Reproduce Issue
I ran it 'locallly':
`salt-call --local saltcheck.run_highstate_tests`

### Versions Report
```
Salt Version:
           Salt: 3000
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.7.3 (default, Dec 20 2019, 18:57:59)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 17.1.2
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist: debian 10.3 
         locale: utf-8
        machine: x86_64
        release: 5.4.0-0.bpo.3-amd64
         system: Linux
        version: debian 10.3 
 
```

**Comment by mchugh19** [comment]
Hi @kiemlicz. Thanks for reporting. This should be corrected in #56327.

**Comment by garethgreenaway** [comment]
@mchugh19 Thanks!
